inst/include/Rcpp/Environment.h

```diff
@@ -393,10 +393,10 @@ namespace Rcpp{
          */
         Environment_Impl new_child(bool hashed) const {
             SEXP newEnvSym = Rf_install("new.env");
-            return Environment_Impl(Rcpp_fast_eval(Rf_lang3(newEnvSym, Rf_ScalarLogical(hashed), Storage::get__()), R_GlobalEnv));
+            Shield<SEXP> call(Rf_lang3(newEnvSym, Rf_ScalarLogical(hashed), Storage::get__()));
```

Comment:
I believe the result of `Rf_ScalarLogical()` needs to be protected as well.

Reply:
Hmm, maybe not -- looks like it's never protected in R's own sources.

Reply:
Can't hurt. Will add it.

Reply:
@kevinushey that's 3 constants: https://github.com/wch/r-source/blob/9f4fd4f2d06f806af32c36e04ec020250242d100/src/include/Rinlinedfuns.h#L990